### DISCUSSION TITLE
sap_host_exporter isn't discovering all the SAP instances

### DISCUSSION THREAD

**Discussion by sericht** [SUSE/sap_host_exporter#62] [Q&A] [answered]
Hello,

I've observed that the sap_host_exporter isn't discovering all the SAP instances of an distributed SAP system. 
I've connected the sap_host_exporter directly to the ASCS instance and the sap_host_exporter is posing only the information about the ASCS instance itself. 
On the same host there are also other instances (PAI) running with different virtual host name (but same SID).

How does the sap_host_exporter installation concept look for an distributed system, do I have to install an sap_host_exporter for each SAP instance, or should it be discovered via the ASCS instance.

Thank you very much in advanced.

Best regards!

**Comment by stefanotorresi** [comment, marked as answer]
Hey @sericht!

Yes, the exporter can only hook into one instance by design, because you will have multiple SAPControl interfaces for each instance (each one listening in their own UDS / TCP port, which are determined by the instance number), so you need to run a separate exporter for each SAP Instance.

The systmd unit file we provide supports multiple exporter service instances OOTB, so you can associate a different configuration file for each exporter instance.
